### DISCUSSION TITLE
(...).collection(...).doc(...).listCollections is not a function

### DISCUSSION THREAD

**Discussion by edulennert** [invertase/react-native-firebase#6240] [Q&A]
Hello guys!

I need a way to list collections that are inside a document. Is there a way to do that usindo react-native-firebase??

Navigating through the firestore documentation I found a function that does exactly that, but it retrieves `.listCollections is not a function` when I try it on rn firebase.

![image](https://user-images.githubusercontent.com/102844542/167765968-fe2bfa0d-6518-44c7-a84a-a8f6fa8b8e78.png)

I don't know if it's a feature that needs to be implemented or a issue of the library. Is there a workaround for this?

**Comment by mikehardy** [comment]
There's a good stackoverflow on that one, I've needed the same. listCollections is a firebase-admin-sdk API I think, which is why it's not accessible here in the client SDKs

https://stackoverflow.com/questions/48258632/fetching-all-collections-in-firestore

I think this is a good take on the 3 different ways to handle it (with first option being "know the collection names ahead of time), 2 being use cloud function to access admin API and 3 being to keep the names of collections in say an array field https://medium.com/firebase-tips-tricks/how-to-list-all-subcollections-of-a-cloud-firestore-document-17f2bb80a166
